src/main/scala/org/broadinstitute/dsde/firecloud/webservice/TrialApiService.scala

```diff
@@ -52,6 +54,24 @@ trait TrialApiService extends HttpService with PerRequestCreator with FireCloudD
             }
           }
         }
+      } ~
+      path("report") {
+        post {
+          requireUserInfo() { userInfo => requestContext =>
+            perRequest(requestContext, TrialService.props(trialServiceConstructor),
+              TrialService.CreateBillingReport(requestContext, userInfo)
+            )
+          }
+        }
+      } ~
+      path("report" / Segment) { (spreadsheetId) =>
+        put {
+          requireUserInfo() { userInfo => requestContext =>
+            perRequest(requestContext, TrialService.props(trialServiceConstructor),
+              TrialService.UpdateBillingReport(requestContext, userInfo, spreadsheetId)
+            )
+          }
+        }
```

Comment:
it looks like this supports the ability to have multiple spreadsheets. Do we really want that? I was picturing a single spreadsheet, and all involved know its location.

Reply:
The issue is that we need to write an initial spreadsheet to *someone's* google drive since FC doesn't have one. That person would then have to be the one to manage sharing. Orch would never change that, unless we want to add that kind of functionality.

Side note - I did think that perhaps the initial campaign manager could supply a list of email addresses that we add as shared users - not sure if that's technically possible yet.